src/changelog/linker/mapper/leadingv.go

```diff
@@ -0,0 +1,82 @@
+package mapper
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/newrelic/release-toolkit/src/changelog"
+	"github.com/newrelic/release-toolkit/src/changelog/linker"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	checkTimeoutSeconds = 1
+)
+
+// LeadingVCheck wraps another mapper and performs a check over the result of Map. If the check is not
+// successful, it tries to generate it prepending a leading v to the version.
+type LeadingVCheck struct {
+	mapper    linker.Mapper
+	checkLink func(link string) bool
+}
+
+// WithLeadingVCheck returns a LeadingVCheck with the provided underlying mapper and a check function which
+// performs a request to the url corresponding to the link and check its status code.
+func WithLeadingVCheck(mapper linker.Mapper) *LeadingVCheck {
+	return &LeadingVCheck{mapper: mapper, checkLink: checkLinkResponse}
+}
+
+func (l *LeadingVCheck) Map(dep changelog.Dependency) string {
+	link := l.mapper.Map(dep)
+	if link == "" || l.checkLink(link) {
+		return link
+	}
+
+	newDep, changed := l.switchDepLeadingV(dep)
+	if !changed {
+		return ""
+	}
+
+	link = l.mapper.Map(newDep)
+	if l.checkLink(link) {
+		return link
+	}
+
+	return ""
+}
+
+func (l *LeadingVCheck) switchDepLeadingV(dep changelog.Dependency) (changelog.Dependency, bool) {
+	literal := dep.To.Original()
+	switchedLiteral := ""
+	if strings.HasPrefix(literal, "v") {
+		switchedLiteral = literal[1:]
+	} else {
+		switchedLiteral = "v" + literal
+	}
+	switchedDep := dep
+	switchedVersion, err := semver.NewVersion(switchedLiteral)
+	if err != nil {
+		log.Debugf("Could not switch leading v to %q valid semver version: %s.", literal, err)
```

Comment:
As this is something that should not be happening, could we promote it to `Errorf` and clarify that is internal?

Reply:
I was going to propose to return error in the second param in this function instead of boolean and handle the log or return in the caller.

Reply:
As it is something that should not happen (if it happens it is because we build a dependency whose `To` field point to an _invalid_ `semver.Version` struct (such as an empty one). I would rather prefer to handle the error here.

Reply:
Changed here: [6f5cbde](https://github.com/newrelic/release-toolkit/pull/134/commits/6f5cbde1ffe7074c0185bed72543d9d8264ec1e3)